app/models/taxon_search_results.rb

```diff
@@ -8,9 +8,12 @@ def initialize(search_response)
   def taxons
     @taxons ||=
       begin
-        search_response['results'].map do |taxon_hash|
+        results = search_response['results'].map do |taxon_hash|
+          next if taxon_hash['publication_state'] == 'unpublished'
```

Comment:
The feature spec exercises this behaviour, but yeah a unit test makes sense. Adding...